src/Plugins/ComplexCore/src/ComplexCore/Filters/ImportCSVDataFilter.cpp

```diff
@@ -433,58 +577,88 @@ IFilter::PreflightResult ImportCSVDataFilter::preflightImpl(const DataStructure&
 {
-  CSVWizardData wizardData = filterArgs.value<CSVWizardData>(k_WizardData_Key);
+  CSVImporterData csvImporterData = filterArgs.value<CSVImporterData>(k_CSVImporterData_Key);
   bool useExistingGroup = filterArgs.value<bool>(k_UseExistingGroup_Key);
   DataPath selectedDataGroup = filterArgs.value<DataPath>(k_SelectedDataGroup_Key);
   DataPath createdDataGroup = filterArgs.value<DataPath>(k_CreatedDataGroup_Key);
 
-  std::string inputFilePath = wizardData.inputFilePath;
-  StringVector headers = wizardData.dataHeaders;
-  DataTypeVector dataTypes = wizardData.dataTypes;
-  CharVector delimiters = wizardData.delimiters;
-  bool consecutiveDelimiters = wizardData.consecutiveDelimiters;
-  usize numLines = wizardData.numberOfLines;
-  usize beginIndex = wizardData.beginIndex;
+  std::string inputFilePath = csvImporterData.inputFilePath;
+  StringVector headers = s_HeaderCache[s_InstanceId].Headers;
+  DataTypeVector dataTypes = csvImporterData.dataTypes;
+  std::vector<bool> skippedArrays = csvImporterData.skippedArrayMask;
+  bool consecutiveDelimiters = csvImporterData.consecutiveDelimiters;
+  usize startImportRow = csvImporterData.startImportRow;
+
+  if(csvImporterData.headerMode == CSVImporterData::HeaderMode::CUSTOM)
+  {
+    headers = csvImporterData.customHeaders;
+  }
 
   DataPath groupPath = createdDataGroup;
   if(useExistingGroup)
   {
     groupPath = selectedDataGroup;
   }
 
-  Result<ParsersVector> parsersResult = createParsers(dataTypes, groupPath, headers, dataStructure);
+  Result<ParsersVector> parsersResult = createParsers(dataTypes, skippedArrays, groupPath, headers, dataStructure);
   if(parsersResult.invalid())
   {
     return ConvertResult(std::move(parsersResult));
   }
 
-  ParsersVector dataParsers = std::move(parsersResult.value());
-
   std::fstream in(inputFilePath.c_str(), std::ios_base::in);
   if(!in.is_open())
   {
     return MakeErrorResult(to_underlying(IssueCodes::FILE_NOT_OPEN), fmt::format("Could not open file for reading: {}", inputFilePath));
   }
 
   // Skip to the first data line
-  skipNumberOfLines(in, beginIndex);
+  if(!skipNumberOfLines(in, startImportRow))
+  {
+    return MakeErrorResult(to_underlying(IssueCodes::CANNOT_SKIP_TO_LINE), fmt::format("Could not skip to the first line in the file to import ({}).", startImportRow));
+  }
+
+  CharVector delimiters;
+  if(csvImporterData.semicolonAsDelimiter)
+  {
+    delimiters.push_back(';');
+  }
+  if(csvImporterData.spaceAsDelimiter)
+  {
+    delimiters.push_back(' ');
+  }
+  if(csvImporterData.commaAsDelimiter)
+  {
+    delimiters.push_back(',');
+  }
+  if(csvImporterData.tabAsDelimiter)
+  {
+    delimiters.push_back('\t');
+  }
 
   float32 threshold = 0.0f;
-  usize numTuples = numLines - beginIndex + 1;
-  for(usize lineNum = beginIndex; lineNum <= numLines; lineNum++)
+  usize numTuples = std::accumulate(csvImporterData.tupleDims.begin(), csvImporterData.tupleDims.end(), 1, std::multiplies<>());
```

Comment:
```suggestion
  usize numTuples = std::accumulate(csvImporterData.tupleDims.cbegin(), csvImporterData.tupleDims.cend(), static_cast<usize>(1), std::multiplies<>());
```
Unfortunately there's a warning that keeps cropping up where `std::accumulate` and other standard library functions deduce the wrong return type if you don't static cast the initial value.